### PR TITLE
[sram_ctrl] Transition into D2

### PR DESCRIPTION
**Pull request by msfschaffner**
Draft PR to prepare moving the SRAM controller into D2.

Signed-off-by: Michael Schaffner <msf@opentitan.org>

### CODE REVIEW

File: hw/ip/sram_ctrl/doc/checklist.md [hunk 1]
```diff
@@ -35,25 +35,25 @@ Code Quality  | [LINT_SETUP][]        | Done        |
...
+Documentation | [MISSING_FUNC][]        | Done        |
+Documentation | [FEATURE_FROZEN][]      | Done        |
+RTL           | [FEATURE_COMPLETE][]    | Done        |
+RTL           | [AREA_CHECK][]          | Done        |
+RTL           | [PORT_FROZEN][]         | Done        |
+RTL           | [ARCHITECTURE_FROZEN][] | Done        |
+RTL           | [REVIEW_TODO][]         | Done        |
+RTL           | [STYLE_X][]             | Done        |
+Code Quality  | [LINT_PASS][]           | Done        |
+Code Quality  | [CDC_SETUP][]           | Not Started | No flow available yet. TBD: Waive?
```
Comment (udinator) [inline hw/ip/sram_ctrl/doc/checklist.md:50]: I think it's ok to waive this until we actually set up the flows and formalize the methodology (e.g. push the doc to the repo, etc...) - IMO we shouldn't block on a nonexistent flow.

**Comment by tjaychen** [comment]
let's discuss this one with Don / Faisal a bit.
Because D2 implies feature freeze and only bug fixes from here on.
I feel like we would need some software feedback to ensure this is okay.

**Comment by msfschaffner** [comment]
Yeah we can hold off on this one until at least the DIF header has been written (implying that the regfile and functionality has been reviewed by SW).

@mateenfaisal, @gkelly, @lenary: who owns the SRAM controller DIF?

**Comment by tjaychen** [comment]
> Yeah we can hold off on this one until at least the DIF header has been written (implying that the regfile and functionality has been reviewed by SW).
> 
> @mateenfaisal, @gkelly, @lenary: who owns the SRAM controller DIF?

yes, but i was actually hoping @mateenfaisal and Don can come up with a process they think works best for the project.

**Comment by msfschaffner** [comment]
@dsandrs

**Comment by mateenfaisal** [comment]
Ack on Sram DIF, process request for new features. Will read this thread.

**Comment by lenary** [comment]
The SRAM DIF is owned by @gkelly - there's no header/implementation for it yet.

**Comment by mateenfaisal** [comment]
Will talk to @gkelly, on when he can get started on SRAM header. For now, I have entered SRAM header as predecessor for SRAM D2 in the plan.

**Comment by gkelly** [comment]
I've taken a look at the docs for this. There are really only two things we really care about from the SW side: calling for a new key and checking the status. We're even going to end up doing those in assembly pre-CRT. This interface looks good to me, I'm comfortable with this going to D2 before we've written the DIF header proper.

**Review by gkelly — APPROVED** [review]

File: hw/ip/sram_ctrl/doc/checklist.md (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/sram_ctrl/doc/checklist.md]: alright, I have waived this item.

**Comment by msfschaffner** [comment]
Thanks for giving this a look @gkelly.
I have waived the CDC lint setup and am putting this up for review.

**Comment by msfschaffner** [comment]
Moving this back to draft due to the missing execute from SRAM scheme.

**Comment by msfschaffner** [comment]
@tjaychen do you think we can move this into D2 now that the execute from SRAM scheme is in place?

**Review by tjaychen — APPROVED** [review]

**Pull request merged**